Fachportal Padagogik.js

```diff
@@ -88,32 +88,78 @@ function doWeb(doc, url) {
 
 function scrape(doc, url) {
 	var m = url.match(/FId=([\w\d]+)(&|$)/);
-	//e.g. http://www.fachportal-paedagogik.de/fis_bildung/suche/fis_ausg.html?FId=A18196&lart=BibTeX&Speichern=Speichern&senden_an=+E-Mail-Adresse
-	var bibUrl = "/fis_bildung/suche/fis_ausg.html?FId=" + m[1] + "&lart=BibTeX";
-	ZU.doGet(bibUrl, function(text){
+	if (m) {
+		//e.g. http://www.fachportal-paedagogik.de/fis_bildung/suche/fis_ausg.html?FId=A18196&lart=BibTeX
+		var bibUrl = "/fis_bildung/suche/fis_ausg.html?FId=" + m[1] + "&lart=BibTeX";
 		var translator = Zotero.loadTranslator("import");
 		translator.setTranslator("9cb70025-a888-4a29-a210-93ec52da40d4");//BibTex translator
-		translator.setString(text);
-		translator.setHandler("itemDone", function(obj, item) {
-			item.attachments.push({
-				title: "Snapshot",
-				document: doc
+		ZU.doGet(bibUrl, function(text){
+			translator.setString(text);
+			translator.setHandler("itemDone", function(obj, item) {
+				finalize(doc, item);
 			});
-			if (item.url) {
+			translator.translate();
+		});
+	} else {
+		var pdfUrl = ZU.xpathText(doc, '//meta[@name="citation_pdf_url"]/@content');
+		Z.debug(pdfUrl);
+
+		/*
+		//fails on http://www.pedocs.de/frontdoor.php?source_opus=9491&la=de
+		Z.debug(ZU.xpathText(doc, '//span[@class="Z3988"]/@title'));
+		var translator = Zotero.loadTranslator("web");
+		translator.setTranslator("05d07af9-105a-4572-99f6-a8e231c0daef");//COinS translator
+		translator.setDocument(doc);//TODO encoding is normally Western and not utf8
+		translator.setHandler("itemDone", function (obj, item) {
+			if (pdfUrl) {
 				item.attachments.push({
-					url: item.url,
-					title: "Link",
-					snapshot: false
+					url: pdfUrl,
+					title: "Full Text PDF",
 				});
-				delete item.url;
-			}
-			if (item.numPages) {
-				item.numPages = parseInt(item.numPages);
 			}
-			item.complete();
+			
+			finalize(doc, item);
 		});
-		translator.translate();			
+		translator.translate();
+		*/
+		
+		var coins = ZU.xpathText(doc, '//span[@class="Z3988"]/@title');
+		if (coins) {
+			//Z.debug(coins);
+			if (coins.indexOf('rft.genre=article') > -1 || coins.indexOf('rft.genre=bookitem') > -1) {//TODO condition here
+				coins = coins.replace('rft.title', 'rft.atitle');
+			}
+			var item = new Zotero.Item();
+			ZU.parseContextObject(coins, item);
```

Comment:
Yes, I think this is preferred, since you probably don't want the translator to handle the entire page anyway (what if there is more than one COinS? You would be missing out on COinS calling the DOI/ISBN translators, but that may also be the source of the issue here.